### ISSUE TITLE
Case insensitive keywords

### ISSUE THREAD

**Issue by iluuu1994** [vimeo/psalm#778]
Psalm looks great! Good work!

Monolog has a few `Boolean` return types (instead of the preferred `bool`):
https://github.com/Seldaek/monolog/blob/master/src/Monolog/Logger.php#L278

```php
$logger->addInfo('...');
```

This causes the following error message:

> Class or interface Monolog\Boolean does not exist

1. Shouldn't this error be suppressed since it's defined in a `vendor` and not my own code?
2. Shouldn't types be case insensitive?

Thanks for your feedback!

**Comment by weirdan** [comment]
- Seldaek/monolog#1037 (there's been no developer activity in the last month as well. To me it seems monolog is ~stable~ dead).
- `Boolean` is a valid classname: https://3v4l.org/YF5SS

> Shouldn't types be case insensitive?

Type could be a class, and classes are case-insensitive only in theory. In common scenarios (e.g. PSR-4, case-sensitive filesystem) user-defined classes are case-sensitive.

As a workaround psalm allows overriding definitions, using [stubs](https://github.com/vimeo/psalm/blob/master/docs/configuration.md#project-settings). Likely you only need to override a couple of classes.

**Comment by muglug** [comment]
@iluuu1994 thanks for the compliments!

As @weirdan said, you could stub.

You can also create a workaround rule:

```php
<UndefinedClass>
  <errorLevel type="suppress">
    <referencedClass name="Monolog\Boolean" />
  </errorLevel>
</UndefinedClass>
```

**Comment by iluuu1994** [comment]
Thanks for the suggestions!

One thing to note is that I'm not even using the return value. Maybe Psalm could just ignore vendor return values unless you use them?

**Comment by muglug** [comment]
Yeah, Psalm warns you about vendor dir issues if they might possibly affect your code. Here, it actually doesn’t, but it’s not hard to imagine a situation where it could blow up even without an assignment (e.g. where an `@return` is missing a required interface)
